### PR TITLE
Ej nc news task6

### PR DESCRIPTION
**Pull request by EdJackson98**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -71,4 +71,45 @@ describe("GET: /api/articles/:article_id", () => {
...
+          expect(response.body.msg).toBe('No article found for article id: 999');
+        });
+    });
+    test('GET:400 sends an appropriate error message when given an invalid ID', () => {
+        return request(app)
+        .get('/api/articles/banana/comments')
+        .expect(400)
+        .then((response) => {
+            expect(response.body.msg).toBe('Invalid ID');
+          });
```
Comment (EddTheDuck) [inline __tests__/app.test.js:118]: test suite isnt complete, need a test for a status 200, valid ID but has no comments

File: __tests__/app.test.js [hunk 2]
```diff
@@ -71,4 +71,45 @@ describe("GET: /api/articles/:article_id", () => {
             expect(response.body.msg).toBe('Invalid ID');
         })
     })
-})
\ No newline at end of file
+})
+
+describe('/api/teams/:article_id/comments', () => {
```
Comment (EddTheDuck) [inline __tests__/app.test.js]: /api/teams ?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -71,4 +71,45 @@ describe("GET: /api/articles/:article_id", () => {
...
+    test('GET:200 sends an array of comments to the user by article_id', () => {
+      return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then((response) => {
+          expect(response.body.comments).toEqual(expect.any(Array));
+          expect(Object.keys(response.body.comments[0])).toEqual(
+            expect.arrayContaining([
```
Comment (EddTheDuck) [inline __tests__/app.test.js]: This is ok I suppose however very much uneeded. You already have a forEach so why not just use your assertions in that ? eg. author: expect.any(String) etc

**Review by EddTheDuck — CHANGES_REQUESTED** [review]
Overall decent but needs more tests, see comments, thanks

**Review by scar1377 — APPROVED** [review]
OK to merge.
One thing to consider - The function checkIDExists is working almost exactly the same as fetchArticleById. Why not make use of fetchArticleById instead of making a new one

**Pull request merged**